libchisel/src/remapstart.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::HashMap;
+
+use super::{ModuleError, ModulePreset, ModuleTranslator};
+use parity_wasm::elements::*;
+
+pub struct RemapStart {}
+
+impl ModulePreset for RemapStart {
+    fn with_preset(preset: &str) -> Result<Self, ()> {
+        match preset {
+            "ewasm" => Ok(RemapStart {}),
+            _ => Err(()),
+        }
+    }
+}
+
+impl ModuleTranslator for RemapStart {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Ok(remap_start(module))
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let mut ret = module.clone();
+        let modified = remap_start(&mut ret);
+        if modified {
+            return Ok(Some(ret));
+        }
+        Ok(None)
+    }
+}
+
+// if the main export doesn't refer to the same function as the start section,
+// point the main export towards the function referenced by the start section
+// remove the start section
+fn replace_export(module: &mut Module, export_name: &str, replacement_func_idx: u32) -> bool {
+    let mut ret = false;
+
+    /*
```

Comment:
Can you get rid of this big comment or is it going to be used?

Reply:
removed.